components/Bookmarkbutton/index.js

```diff
@@ -1,29 +1,25 @@
-import { useState, useEffect } from "react";
+import styled from "styled-components";
 
-export default function BookmarkButton({ itemId, onBookmark }) {
-  const [isBookmarked, setIsBookmarked] = useState(false);
-
-  useEffect(() => {
-    const favorites = JSON.parse(localStorage.getItem("favorites")) || [];
-    setIsBookmarked(favorites.includes(itemId));
-  }, [itemId]);
+const StyledButton = styled.button`
+  /* Hier kannst du das Styling des Buttons anpassen */
+`;
 
+export default function BookmarkButton({ itemId, isBookmarked, onBookmark }) {
   const handleClick = () => {
-    setIsBookmarked(!isBookmarked);
-    onBookmark(itemId, !isBookmarked);
+    onBookmark(!isBookmarked);
   };
 
   return (
-    <button onClick={handleClick}>
+    <StyledButton onClick={handleClick}>
       {isBookmarked ? (
         <svg
           xmlns="http://www.w3.org/2000/svg"
           viewBox="0 0 24 24"
           width="24"
           height="24"
         >
-          <title>bookmark</title>
-          <path d="M17,3H7A2,2 0 0,0 5,5V21L12,18L19,21V5A2,2 0 0,0 17,3Z" />
+          <title>star</title>
+          <path d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z" />
         </svg>
```

Comment:
You could consider creating an external svg file and save your svg code there and then import it as component. You could reduce your code significantly🤓